src/components/modal/__internal__/modal-manager.ts

```diff
@@ -10,6 +10,10 @@ class ModalManagerInstance {
   private modalList: ModalList;
 
   constructor() {
+    if (typeof window === "undefined") {
```

Comment:
`typeof window === "undefined"` is the safe way to check for the existence of `window` in Node.js environments.